### ISSUE TITLE
CloudStack RHEL8 images built in EKS-A pipeline are not working with EKS-A CLI

### ISSUE THREAD

**Issue by maxdrib** [aws/eks-anywhere#4282]
**What happened**:
Attempted to create a CloudStack cluster with EKS-A using a RHEL image built in the EKS-A pipeline and the control plane was not able to start, using the `TestCloudStackKubernetes121SimpleFlow` e2e test. When using unstacked etcd, the etcd VM fails to start successfully.

**What you expected to happen**:
I expected the cluster to be created successfully.

**How to reproduce it (as minimally and precisely as possible)**:
Import the latest CloudStack 1.21 image built in the EKS-A pipeline (https://projectbuildpipeline-857-pipelineoutputartifactsb-10ajmk30khe3f.s3.us-west-2.amazonaws.com/projects/kubernetes-sigs/image-builder/1-21/cloudstack/redhat/latest/redhat.qcow2) into a CloudStack environment and attempt to run the `TestCloudStackKubernetes121SimpleFlow` e2e test. 

**Anything else we need to know?**:
* In working with this situation, we have identified a possible bug with cloud-init/RHEL8.7 compatibility, as documented: https://bugs.launchpad.net/cloud-init/+bug/1998655

**Environment**:
- EKS Anywhere Release: main
- EKS Distro Release: latest

**Comment by maxdrib** [comment]
Looking at the way these systemd files are generated in the [cloud-init source code](https://git.launchpad.net/cloud-init/tree/systemd/cloud-init.service.tmpl#n15), it seems we should be able to configure the addition of this NetworkManager `After=` entry by passing in the correct OS variant. I suspect EKS-A is not correctly plumbing the `rhel` OS variant when building these templates.

**Comment by maxdrib** [comment]
Adding some additional details here, we were able to identify that this issue is not just limited to RHEL8.7, but also can be observed in RHEL8.6. The root cause emerges from the cloud-init source code with rhel. Looking at cloud-init releases for rhel in https://git.centos.org/rpms/cloud-init/releases, we saw the introduction of this systemd After= condition [here](https://git.centos.org/rpms/cloud-init/blob/e2481008492b618ff37907f88dc718d0a2154724/f/SOURCES/0007-Remove-race-condition-between-cloud-init-and-Network.patch) to prevent the NetworkManager/cloud-init race condition, and its subsequent removal [here](https://git.centos.org/rpms/cloud-init/blob/6af34fd764c66bb42d83d6fb706b92a1cd50cec1/f/SOURCES/ci-Remove-rhel-specific-files.patch). We have identified the issue to affect cloud-init versions >cloud-init-22.1-1.el8 (https://git.centos.org/rpms/cloud-init/c/1992bd68fc512e22df783ff955edccbad66dbd60?branch=c8s). We recommend updating the systemd service template for cloud-init to account for this additional After= condition.
